common/djangoapps/student/tests/tests.py

```diff
@@ -404,6 +414,91 @@ def test_refundable_when_certificate_exists(self):
         self.assertFalse(enrollment.refundable())
 
     @unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+    def test_refundable_when_refund_window(self):
+        CourseModeFactory.create(
+            course_id=self.course.id,
+            mode_slug='verified',
+            mode_display_name='Verified',
+            expiration_datetime=datetime.now(pytz.UTC) + timedelta(days=1)
+        )
+
+        enrollment = CourseEnrollment.enroll(self.user, self.course.id, mode='verified')
+
+        self.assertTrue(enrollment.refundable())
+
+        with patch('student.models.CourseEnrollment._refund_window_end_date') as window:
+            window.return_value = datetime.now() - timedelta(days=1)
+            self.assertFalse(enrollment.refundable())
+
+            window.return_value = datetime.now() + timedelta(days=1)
+            self.assertTrue(enrollment.refundable())
+
+    @ddt.data(
+        (timedelta(days=1), timedelta(days=2), timedelta(days=2), 14),
+        (timedelta(days=2), timedelta(days=1), timedelta(days=2), 14),
+        (timedelta(days=1), timedelta(days=2), timedelta(days=2), 1),
+        (timedelta(days=2), timedelta(days=1), timedelta(days=2), 1),
+    )
+    @ddt.unpack
+    @httpretty.activate
+    @override_settings(ECOMMERCE_API_SIGNING_KEY=TEST_API_SIGNING_KEY, ECOMMERCE_API_URL=TEST_API_URL)
+    @unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+    def test_refund_window_end_date(self, order_date_delta, course_start_delta, expected_date_delta, days):
+        """ Assert that the later refund date is and config window timedelta are used to calculate window end date."""
```

Comment:
We are also testing that it does the right thing for multiple refund periods.  The first two test that we get the expected result using the default 14 days. The last two use a period of 1 day to ensure that we get the correct results when using a non-default